demo/Render.hpp

```diff
@@ -0,0 +1,841 @@
+*/
+#ifndef PEGASUS_RENDER_HPP
+#define PEGASUS_RENDER_HPP
+
+#include <glbinding/gl44ext/gl.h>
+#define GLFW_INCLUDE_NONE
+#include <GLFW/glfw3.h>
+#include <glm/ext.hpp>
+
+#include <vector>
+
+using namespace gl;
+
+namespace pegasus
+{
+namespace render
+{
+using Handle = uint32_t;
+
+namespace mesh
+{
+/**
+ * @brief Stores mesh data and GPU OpenGL handle information
+ */
+struct Mesh
+{
+    std::vector<GLdouble> vertices;
+    std::vector<GLuint> indices;
+    glm::mat4 model = glm::mat4(1);
+    glm::vec3 color = glm::vec3(1, 0, 0);
+
+    /**
+     * @brief Stores OpenGL vertex array and vertex buffer handles
+     */
+    struct BufferData
+    {
+        GLuint vertexArrayObject = 0;
+        GLuint elementBufferObject = 0;
+        GLuint vertexBufferObject = 0;
+    };
+    BufferData bufferData;
+};
+
+/**
+ * @brief Copies mesh data to the GPU memory and updates mesh with its handles
+ * @param[in,out] mesh holds vertex data and opengl handles
+ */
+void Allocate(Mesh& mesh);
+
+/**
+ * @brief Removes given mesh from the GPU memory and updates its handles
+ * @param[in,out] mesh holds vertex data and opengl handles
+ */
+void Deallocate(Mesh& mesh);
+
+/**
+ * @brief Makes mesh and copies its data to the GPU memory
+ * @param[in] vertices vertex data
+ * @param[in] indices index data
+ * @return created mesh
+ */
+Mesh Create(std::vector<GLdouble>&& vertices, std::vector<GLuint>&& indices);
+
+/**
+ * @brief Makes line segment mesh and copies its data to the GPU memory
+ * @param[in] start first point of the line segment
+ * @param[in] end second point of the line segment
+ * @return created mesh
+ */
+Mesh CreateLineSegment(glm::vec3 start, glm::vec3 end);
+
+/**
+ * @brief Makes plane mesh and copies its data to the GPU memory
+ * @param[in] normal plane normal
+ * @param[in] length length of the side of the plane
+ * @return created mesh
+ */
+Mesh CreatePlane(glm::dvec3 normal, double length);
+
+/**
+ * @brief Makes ICO sphere mesh and copies its data to the GPU memory
+ * @param[in] radius sphere's radius
+ * @param[in] depth sphere's level of the details
+ * @return created mesh
+ */
+Mesh CreateSphere(double radius, uint32_t depth);
+
+/**
+ * @brief Makes box mesh and copies its data to the GPU memory
+ * @param[in] i orthogonal basis vector of the box
+ * @param[in] j orthogonal basis vector of the box
+ * @param[in] k orthogonal basis vector of the box
+ * @return created mesh
+ */
+Mesh CreateBox(glm::dvec3 i, glm::dvec3 j, glm::dvec3 k);
+
+/**
+ * @brief Deallocates GPU and local memory used by the mesh
+ * @param[in] mesh reference to the mesh
+ */
+void Delete(Mesh& mesh);
+} // namespace mesh
+
+namespace asset
+{
+/**
+ * @brief Represents a unit of storage with id
+ * @tparam T storage data type
+ */
+template < typename T >
+struct Asset
+{
+    Handle id;
+    T data;
+};
+
+/**
+ * @brief Finds first available or creates new element for the given asset vector and returns its handle
```

Comment:
This isn't really a brief. You could use this as main description and leave a brief with something shorter

```cpp
/** @brief Returns a handle for @p T data
 *
 *  Searches for an empty handle in @p data. If there are no empty handles, creates a new one.
 *
 *  [...]
 */
```